app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt

```diff
@@ -349,6 +361,11 @@ class MessageComposerViewModel @Inject constructor(
         enqueueMessageSelfDeletionUseCase(conversationId, uiMessage.header.messageId)
     }
 
+    fun updateSelfDeletingMessages(newSelfDeletingStatus: SelfDeletingMessagesStatus) = viewModelScope.launch {
+        persistNewSelfDeletingStatus(newSelfDeletingStatus)
+        messageComposerViewState = messageComposerViewState.copy(selfDeletingMessagesStatus = newSelfDeletingStatus)
```

Comment:
do we really need to do this : 
messageComposerViewState = messageComposerViewState.copy(selfDeletingMessagesStatus = newSelfDeletingStatus) ? 

Isn't
 private fun observeSelfDeletingMessagesStatus() = viewModelScope.launch {
        observeSelfDeletingMessages().collect { selfDeletingStatus ->
            messageComposerViewState = messageComposerViewState.copy(selfDeletingMessagesStatus = selfDeletingStatus)
        }
    }
    
    suppsed to do the same ?

Reply:
True story.